### ISSUE TITLE
Integration tests with iceoryx related projects

### ISSUE THREAD

**Issue by elfenpiff** [eclipse-iceoryx/iceoryx#365]
## Brief feature description

A project like [rmw_iceoryx](github.com/ros2/rmw_iceoryx/) could break when we change our public API in master. 
Here we could either document the best practice that you only work with tags but its maybe not the best approach for projects which are so closely related to iceoryx like `rmw_iceoryx`.

I would suggest to introduce some "iceoryx master" projects where we add a job in our CI pipeline. Here we could:
  - fail if the "iceoryx master" project is not building
     - I think here we would require committer rights for the project otherwise this could block us hard
  - inform the code owner that with the new iceoryx version his master will not work anymore

**Comment by elfenpiff** [comment]
@Karsten1987 @elBoberido @mossmaurice @budrus @dkroenke whats your opinion?

**Comment by elBoberido** [comment]
@elfenpiff my opinion is to not break the API in the first place ... at least not after 1.0 release. We need some checks for this and building rmw_iceoryx could be one of them.

If someone accesses private API it's not our problem. We should be willing to extend our public API in such cases, though.

**Comment by budrus** [comment]
I guess if we have long term support releases with a stable API people would more use these. E.g. I don't think that people besides the ones who develop new stuff work on master of ROS. There are frequent releases with clear support timeline. 
In rmw_iceoryx we already have a CI job that checks compatibility with iceoryx. Having it on this side makes more sense for me. There we can have a "check the masters" action for information. But we should push the users more to the stable releases. 

What you propose is a third project that uses rmw_iceoryx and iceoryx and checks these together?

**Comment by elBoberido** [comment]
@budrus I smell a `iceoryx ecosystem health` badge 😁

**Comment by budrus** [comment]
When we are talking about a badge I'm in ;-)

**Comment by elfenpiff** [comment]
@budrus I suggest the following badge: https://upload.wikimedia.org/wikipedia/commons/8/83/DEA_badge_C.PNG
It looks epic - but of course with an oryx instead of an eagle!

**Comment by elfenpiff** [comment]
@elBoberido @dkroenke @FerdinandSpitzschnueffler I recommend the following iceoryx developer badge: https://media.gettyimages.com/photos/police-badge-picture-idAA017580?s=612x612

But with "Ice0ryx Developer" instead of police officer!

**Comment by elBoberido** [comment]
@elfenpiff we are talking more about this https://shields.io/category/build ;)

**Comment by elfenpiff** [comment]
Yeah but I was talking about real badges! Additionally, every developer should get a small mission accomplished badge for every issue which the developer is closing.

I would like to show my accomplishments like here: https://www.snopes.com/tachyon/2016/05/nkextramedals.jpg?resize=733,412
